In [1]:
from inference_and_generation import *
from evaluate_pass_k import compute_metrics


/cephfs/users/bashir/RCoT-Diffusion-LLM/.venv/lib/python3.11/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


Using cuda:0


In [2]:
%%time

model_name = "Qwen/Qwen2.5-7B"
data = "gsm8k"
num_evals_to_use = 5
few_shot = 0
batch_size = 2
gen_length = 256
temperature = 0.0
top_p = 0.95
n_samples = 2

metrics = evaluate_auto_regressive_model(
    model_name, data, num_evals_to_use, few_shot, batch_size, gen_length, temperature, top_p, n_samples
    )

Loading checkpoint shards: 100%|██████████| 4/4 [00:00<00:00, 104.61it/s]


evaluating 5 examples
Model type: Base


Evaluating (total samples: 5):   0%|          | 0/3 [00:00<?, ?it/s]The following generation flags are not valid and may be ignored: ['temperature', 'top_p']. Set `TRANSFORMERS_VERBOSITY=info` for more details.
The following generation flags are not valid and may be ignored: ['temperature', 'top_p']. Set `TRANSFORMERS_VERBOSITY=info` for more details.
Evaluating (total samples: 5):  33%|███▎      | 1/3 [00:14<00:28, 14.46s/it]The following generation flags are not valid and may be ignored: ['temperature', 'top_p']. Set `TRANSFORMERS_VERBOSITY=info` for more details.
The following generation flags are not valid and may be ignored: ['temperature', 'top_p']. Set `TRANSFORMERS_VERBOSITY=info` for more details.
Evaluating (total samples: 5):  67%|██████▋   | 2/3 [00:30<00:15, 15.58s/it]The following generation flags are not valid and may be ignored: ['temperature', 'top_p']. Set `TRANSFORMERS_VERBOSITY=info` for more details.
The following generation flags are not valid and may be ignored: 

Saved generations to results/Qwen_Qwen2.5-7B-base_256_0_2_0.0_generations_ar.json
CPU times: user 35.1 s, sys: 11.5 s, total: 46.6 s
Wall time: 55.6 s


In [3]:
%%time

diffusion_model_name = "GSAI-ML/LLaDA-8B-Base"
data = "gsm8k"
num_evals_to_use = 5
few_shot = 0
batch_size = 2
gen_length = 256
diffusion_steps = 32
temperature = 0.0
cfg_scale = 0.0
steps = 32
block_length = 32
remasking = "low_confidence"
alg = "entropy"
alg_temp = 0.0
top_p = 0.95
top_k = None
n_samples = 2

metrics = evaluate_dllm(
    diffusion_model_name, data, num_evals_to_use, few_shot, batch_size, gen_length, diffusion_steps, temperature, cfg_scale, steps, block_length, remasking, alg, alg_temp, top_p, top_k, n_samples
    )

Loading checkpoint shards: 100%|██████████| 6/6 [00:00<00:00, 176.15it/s]


evaluating 5 examples
Model type: Base


Evaluating (total samples: 5): 100%|██████████| 3/3 [00:18<00:00,  6.10s/it]


Saved generations to results/GSAI-ML_LLaDA-8B-Base_256_32_0_2_0.0_generations_testing.json
CPU times: user 22.3 s, sys: 13.3 s, total: 35.6 s
Wall time: 39.4 s


In [4]:
%%time

diffusion_model_name = "GSAI-ML/LLaDA-8B-Base"
data = "gsm8k"
num_evals_to_use = 5
few_shot = 0
batch_size = 2
gen_length = 256
diffusion_steps = 64
temperature = 0.0
cfg_scale = 0.0
steps = 64
block_length = 32
remasking = "low_confidence"
alg = "entropy"
alg_temp = 0.0
top_p = 0.95
top_k = None
n_samples = 2

metrics = evaluate_dllm(
    diffusion_model_name, data, num_evals_to_use, few_shot, batch_size, gen_length, diffusion_steps, temperature, cfg_scale, steps, block_length, remasking, alg, alg_temp, top_p, top_k, n_samples
    )

Loading checkpoint shards: 100%|██████████| 6/6 [00:00<00:00, 174.36it/s]


evaluating 5 examples
Model type: Base


Evaluating (total samples: 5): 100%|██████████| 3/3 [00:37<00:00, 12.58s/it]


Saved generations to results/GSAI-ML_LLaDA-8B-Base_256_64_0_2_0.0_generations_testing.json
CPU times: user 39.2 s, sys: 15.4 s, total: 54.5 s
Wall time: 58 s


In [2]:
%%time

diffusion_model_name = "GSAI-ML/LLaDA-8B-Base"
data = "gsm8k"
num_evals_to_use = 5
few_shot = 0
batch_size = 2
gen_length = 256
diffusion_steps = 256
temperature = 0.0
cfg_scale = 0.0
steps = 256
block_length = 32
remasking = "low_confidence"
alg = "entropy"
alg_temp = 0.0
top_p = 0.95
top_k = None
n_samples = 2
device = "cuda"
output_dir = "results"

metrics = evaluate_dllm(
    diffusion_model_name, data, num_evals_to_use, few_shot, batch_size, gen_length, diffusion_steps, temperature, cfg_scale, steps, block_length, remasking, alg, alg_temp, top_p, top_k, n_samples, device, output_dir
    )

Loading checkpoint shards: 100%|██████████| 6/6 [00:00<00:00, 174.21it/s]


evaluating 5 examples
Model type: Base


Evaluating (total samples: 5): 100%|██████████| 3/3 [02:27<00:00, 49.05s/it]


Saved generations to results/GSAI-ML_LLaDA-8B-Base_256_256_0_2_0.0_generations_testing.json
CPU times: user 2min 17s, sys: 25.8 s, total: 2min 43s
Wall time: 2min 46s


In [2]:
%%time

diffusion_model_name = "GSAI-ML/LLaDA-8B-Base"
data = "gsm8k"
num_evals_to_use = 5
few_shot = 0
batch_size = 2
gen_length = 256
diffusion_steps = 256
temperature = 0.0
cfg_scale = 0.0
steps = 256
block_length = 32
remasking = "low_confidence"
alg = "entropy"
alg_temp = 0.0
top_p = 0.95
top_k = None
n_samples = 2
use_cache=True
dual_cache=True
threshold = 0.3
factor = 1.0

metrics = evaluate_fast_dllm(
    diffusion_model_name, data, num_evals_to_use, few_shot, batch_size, gen_length, diffusion_steps, temperature, cfg_scale, steps, block_length, remasking, alg, alg_temp, top_p, top_k, n_samples, use_cache, dual_cache, threshold, factor
    )

Loading checkpoint shards: 100%|██████████| 6/6 [00:00<00:00, 169.58it/s]


evaluating 5 examples
Model type: Base


Evaluating (total samples: 5): 100%|██████████| 3/3 [00:23<00:00,  7.91s/it]


Saved generations to results/GSAI-ML_LLaDA-8B-Base_256_256_0_2_0.0_generations_testing_fast_dllm.json
CPU times: user 25.9 s, sys: 11.6 s, total: 37.6 s
Wall time: 42.2 s



## Results Summary Table

| # | Method | Model | Steps | Gen Length | Block | Wall Time (s) | Time/Sample (s) | Time/Generation (s) | Throughput (gen/s) | Output Quality | Status |
|---|--------|-------|-------|------------|-------|---------------|-----------------|---------------------|-------------------|----------------|--------|
| 1 | **Official LLaDA** | LLaDA-8B-Base | **256** | 256 | 32 | 49.05 | 4.90 | 2.45 | 0.41 | ✅ **Excellent** | **WORKING** |
| 2 | **Fast-dLLM** | LLaDA-8B-Base | 256 | 256 | 32 | 7.90 | 0.79 | 0.40 | 2.53 | ⚠️ Good | Minor Issues |
| 3 | **Official LLaDA** | LLaDA-8B-Base | 64 | 256 | 32 | 12.58 | 1.26 | 0.63 | 1.59 | ❌ Failed | Zeros/incomplete |
| 4 | **Official LLaDA** | LLaDA-8B-Base | 32 | 256 | 32 | 6.09 | 0.61 | 0.30 | 3.28 | ❌ Failed | Completely garbled |
| 5 | **Autoregressive** | Qwen2.5-7B | N/A | 256 | N/A | 11.37 | 1.14 | 0.57 | 1.76 | ✅ **Excellent** | **WORKING** |

In [5]:
%%time

diffusion_model_name = "Dream-org/Dream-v0-Base-7B"
data = "gsm8k"
num_evals_to_use = 5
few_shot = 0
batch_size = 2
gen_length = 256
diffusion_steps = 32
temperature = 0.0
cfg_scale = 0.0
steps = 32
block_length = 32
remasking = "low_confidence"
alg = "entropy"
alg_temp = 0.0
top_p = 0.95
top_k = None
n_samples = 2

metrics = evaluate_dllm(
    diffusion_model_name, data, num_evals_to_use, few_shot, batch_size, gen_length, diffusion_steps, temperature, cfg_scale, steps, block_length, remasking, alg, alg_temp, top_p, top_k, n_samples
    )

Loading checkpoint shards: 100%|██████████| 4/4 [00:03<00:00,  1.04it/s]


evaluating 5 examples
Model type: Base


Evaluating (total samples: 5): 100%|██████████| 3/3 [00:13<00:00,  4.57s/it]


Saved generations to results/Dream-org_Dream-v0-Base-7B_256_32_0_2_0.0_generations_testing.json
CPU times: user 16.7 s, sys: 17.1 s, total: 33.8 s
Wall time: 51.3 s


In [6]:
%%time

diffusion_model_name = "Dream-org/Dream-v0-Base-7B"
data = "gsm8k"
num_evals_to_use = 5
few_shot = 0
batch_size = 2
gen_length = 256
diffusion_steps = 64
temperature = 0.0
cfg_scale = 0.0
steps = 64
block_length = 32
remasking = "low_confidence"
alg = "entropy"
alg_temp = 0.0
top_p = 0.95
top_k = None
n_samples = 2

metrics = evaluate_dllm(
    diffusion_model_name, data, num_evals_to_use, few_shot, batch_size, gen_length, diffusion_steps, temperature, cfg_scale, steps, block_length, remasking, alg, alg_temp, top_p, top_k, n_samples
    )

Loading checkpoint shards: 100%|██████████| 4/4 [00:00<00:00, 102.54it/s]


evaluating 5 examples
Model type: Base


Evaluating (total samples: 5): 100%|██████████| 3/3 [00:30<00:00, 10.06s/it]


Saved generations to results/Dream-org_Dream-v0-Base-7B_256_64_0_2_0.0_generations_testing.json
CPU times: user 32.5 s, sys: 9.74 s, total: 42.2 s
Wall time: 46.8 s


In [7]:
%%time

diffusion_model_name = "Dream-org/Dream-v0-Base-7B"
data = "gsm8k"
num_evals_to_use = 5
few_shot = 0
batch_size = 2
gen_length = 256
diffusion_steps = 256
temperature = 0.0
cfg_scale = 0.0
steps = 256
block_length = 32
remasking = "low_confidence"
alg = "entropy"
alg_temp = 0.0
top_p = 0.95
top_k = None
n_samples = 2

metrics = evaluate_dllm(
    diffusion_model_name, data, num_evals_to_use, few_shot, batch_size, gen_length, diffusion_steps, temperature, cfg_scale, steps, block_length, remasking, alg, alg_temp, top_p, top_k, n_samples
    )

Loading checkpoint shards: 100%|██████████| 4/4 [00:00<00:00, 95.21it/s]


evaluating 5 examples
Model type: Base


Evaluating (total samples: 5): 100%|██████████| 3/3 [01:59<00:00, 39.73s/it]


Saved generations to results/Dream-org_Dream-v0-Base-7B_256_256_0_2_0.0_generations_testing.json
CPU times: user 2min 1s, sys: 10.8 s, total: 2min 12s
Wall time: 2min 18s


In [14]:
%%time

diffusion_model_name = "Dream-org/Dream-v0-Base-7B"
data = "gsm8k"
num_evals_to_use = 5
few_shot = 0
batch_size = 1
gen_length = 256
diffusion_steps = 256
temperature = 0.0
cfg_scale = 0.0
steps = 256
block_length = 32
remasking = "low_confidence"
alg = "confidence_threshold"
alg_temp = 0.0
top_p = 0.95
top_k = None
n_samples = 2
use_cache=True
dual_cache=True
threshold = 0.75
factor = 1.0

metrics = evaluate_fast_dllm(
    diffusion_model_name, data, num_evals_to_use, few_shot, batch_size, gen_length, diffusion_steps, temperature, cfg_scale, steps, block_length, remasking, alg, alg_temp, top_p, top_k, n_samples, use_cache, dual_cache, threshold, factor
    )

Loading checkpoint shards: 100%|██████████| 4/4 [00:00<00:00, 99.64it/s]


evaluating 5 examples
Model type: Base


Evaluating (total samples: 5): 100%|██████████| 5/5 [00:33<00:00,  6.63s/it]


Saved generations to results/Dream-org_Dream-v0-Base-7B_256_256_0_2_0.0_generations_testing_fast_dllm.json
CPU times: user 35.2 s, sys: 9.64 s, total: 44.9 s
Wall time: 49.6 s


## Results Summary Table (End-to-End Times)

| # | Method | Model | Steps | Temp | Block | Cache | Alg | Wall Time (s) | Time/Sample (s) | Time/Gen (s) | Throughput (tok/s) | Output Quality | Status |
|---|--------|-------|-------|------|-------|-------|-----|---------------|-----------------|-----------------|-------------------|----------------|--------|
| 1 | **Dream Official** | Dream-7B | 256 | 0.0 | N/A | ❌ No | entropy | **138.0s** | 13.80 | 6.90 | 37.1 tok/s | ⚠️ Mixed | Incomplete |
| 2 | **Dream Official** | Dream-7B | 64 | 0.0 | N/A | ❌ No | entropy | **46.8s** | 4.68 | 2.34 | 109.4 tok/s | ⚠️ Mixed | Incomplete |
| 3 | **Dream Official** | Dream-7B | 32 | 0.0 | N/A | ❌ No | entropy | **51.3s** | 5.13 | 2.57 | 99.9 tok/s | ❌ Failed | Loading overhead |
| 4 | **Fast-dLLM** | Dream-7B | 256 | 0.0 | 32 | ✅ Dual | entropy | **49.6s** | 4.96 | 2.48 | 103.2 tok/s | ❌ Poor | Mask tokens |
| 5 | **Autoregressive** | Qwen-7B | N/A | 0.0 | N/A | ❌ No | N/A | **55.6s** | 5.56 | 2.78 | 92.1 tok/s | ✅ **Perfect** | **WORKING** |

**Note**: These are end-to-end times including model loading overhead

In [3]:
%%time

diffusion_model_name = "Dream-org/Dream-v0-Base-7B"
data = "gsm8k"
num_evals_to_use = 256
few_shot = 4
batch_size = 1
gen_length = 256
diffusion_steps = 256
temperature = 0.0
cfg_scale = 0.0
steps = 256
block_length = 32
remasking = "low_confidence"
alg = "confidence_threshold"
alg_temp = 0.0
top_p = 0.95
top_k = None
n_samples = 1
use_cache=True
dual_cache=True
threshold = 0.75
factor = 1.0

metrics = evaluate_fast_dllm(
    diffusion_model_name, data, num_evals_to_use, few_shot, batch_size, gen_length, diffusion_steps, temperature, cfg_scale, steps, block_length, remasking, alg, alg_temp, top_p, top_k, n_samples, use_cache, dual_cache, threshold, factor
    )

Loading checkpoint shards: 100%|██████████| 4/4 [00:00<00:00, 103.34it/s]


Created 4 few-shot examples
evaluating 256 examples
Model type: Base


Evaluating (total samples: 256): 100%|██████████| 256/256 [23:46<00:00,  5.57s/it]


Saved generations to results/Dream-org_Dream-v0-Base-7B_256_256_4_1_0.0_generations_testing_fast_dllm.json
CPU times: user 23min 47s, sys: 10.4 s, total: 23min 57s
Wall time: 24min 6s


In [4]:

compute_metrics(
    results_file="results/Dream-org_Dream-v0-Base-7B_256_256_4_1_0.0_generations_testing_fast_dllm.json",
    samples_per_problem=10,
    k_values=[1, 5, 10]
)

Loaded 256 generations

--- Answer Format Analysis ---

Format Distribution:
  float     :  248 ( 96.9%)
  int       :    0 (  0.0%)
  list      :    0 (  0.0%)
  string    :    8 (  3.1%)
  none      :    0 (  0.0%)

Example Formats:

  FLOAT examples:
    1. Q: Janet had 22 green pens and 10 yellow pens. Then she bought 6 bags of blue pens ...
       A: 98.0
    2. Q: Together, Sofie, Anne, and Fawn have 85 books. If Sofie has 25 more books than A...
       A: 60.0

  STRING examples:
    1. Q: Christina records her mood every day on a calendar. Over the past thirty days of...
       A: <reasoning>
Christina had 12 + 8 + 8 = <<12+8+8=28>>28 days of good, bad, and neutral moods.
She had 30 - 28 = <<30-28=2>>2 days of good, bad, and neutral moods.
She had 8 + 8 + 8 = <<8+8+8=24>>24 day
    2. Q: As Sally walked to school, she was holding the strings to 25 red balloons, 7 gre...
       A: <reasoning>
Sally had 25 red balloons, 7 green balloons, and 12 yellow balloons. 40% of the red bal

{'pass_at_k': {1: 0.72265625, 5: 0.0, 10: 0.0},
 'accuracy': 0.72265625,
 'avg_correct_per_problem': 0.72265625,
 'correct_rate': 0.72265625,
 'problems_evaluated': 256,
 'total_correct': 185,
 'total_samples': 256}

In [5]:
%%time

diffusion_model_name = "GSAI-ML/LLaDA-8B-Base"
data = "gsm8k"
num_evals_to_use = 256
few_shot = 4
batch_size = 1
gen_length = 256
diffusion_steps = 256
temperature = 0.0
cfg_scale = 0.0
steps = 256
block_length = 32
remasking = "low_confidence"
alg = "entropy"
alg_temp = 0.0
top_p = 0.95
top_k = None
n_samples = 1
use_cache=True
dual_cache=True
threshold = 0.3
factor = 1.0

metrics = evaluate_fast_dllm(
    diffusion_model_name, data, num_evals_to_use, few_shot, batch_size, gen_length, diffusion_steps, temperature, cfg_scale, steps, block_length, remasking, alg, alg_temp, top_p, top_k, n_samples, use_cache, dual_cache, threshold, factor
    )

Loading checkpoint shards: 100%|██████████| 6/6 [00:00<00:00, 168.14it/s]


Created 4 few-shot examples
evaluating 256 examples
Model type: Base


Evaluating (total samples: 256): 100%|██████████| 256/256 [13:49<00:00,  3.24s/it]


Saved generations to results/GSAI-ML_LLaDA-8B-Base_256_256_4_1_0.0_generations_testing_fast_dllm.json
CPU times: user 13min 53s, sys: 11.2 s, total: 14min 4s
Wall time: 14min 11s


In [6]:

compute_metrics(
    results_file="results/GSAI-ML_LLaDA-8B-Base_256_256_4_1_0.0_generations_testing_fast_dllm_threshold_0.3.json",
    samples_per_problem=10,
    k_values=[1, 5, 10]
)

Loaded 256 generations

--- Answer Format Analysis ---

Format Distribution:
  float     :  176 ( 68.8%)
  int       :    0 (  0.0%)
  list      :    0 (  0.0%)
  string    :   80 ( 31.2%)
  none      :    0 (  0.0%)

Example Formats:

  FLOAT examples:
    1. Q: A loaf of bread at the bakery costs $2. Bagels cost $1 each. How much more do 3 ...
       A: 60.0
    2. Q: Tim decides to cancel his cable subscription and get streaming services.  He get...
       A: 44.0

  STRING examples:
    1. Q: The local firefighters are doing a “fill the boot” fundraiser. Their goal is to ...
       A: <reasoning>
They have to raise $6300 - $2000 = <<6000-2000=4300>>4300 more

They need to raise $43300 / $100 = <<3300/100=33>>33 more hours

They will need to raiseise 33 hours + 3 hours = <<33+3=36>>
    2. Q: If Marcy works for the same company for 40 years, she gets an annual pension of ...
       A: <reasoning>
First, we need to find how many years Marcy has worked before she gets the 5% reduction

{'pass_at_k': {1: 0.30859375, 5: 0.0, 10: 0.0},
 'accuracy': 0.30859375,
 'avg_correct_per_problem': 0.30859375,
 'correct_rate': 0.30859375,
 'problems_evaluated': 256,
 'total_correct': 79,
 'total_samples': 256}

In [7]:
%%time

diffusion_model_name = "GSAI-ML/LLaDA-8B-Base"
data = "gsm8k"
num_evals_to_use = 256
few_shot = 4
batch_size = 1
gen_length = 256
diffusion_steps = 256
temperature = 0.0
cfg_scale = 0.0
steps = 256
block_length = 32
remasking = "low_confidence"
alg = "entropy"
alg_temp = 0.0
top_p = 0.95
top_k = None
n_samples = 1
use_cache=True
dual_cache=True
threshold = 0.5
factor = 1.0

metrics = evaluate_fast_dllm(
    diffusion_model_name, data, num_evals_to_use, few_shot, batch_size, gen_length, diffusion_steps, temperature, cfg_scale, steps, block_length, remasking, alg, alg_temp, top_p, top_k, n_samples, use_cache, dual_cache, threshold, factor
    )

Loading checkpoint shards: 100%|██████████| 6/6 [00:00<00:00, 175.64it/s]


Created 4 few-shot examples
evaluating 256 examples
Model type: Base


Evaluating (total samples: 256): 100%|██████████| 256/256 [18:52<00:00,  4.42s/it]


Saved generations to results/GSAI-ML_LLaDA-8B-Base_256_256_4_1_0.0_generations_testing_fast_dllm.json
CPU times: user 18min 57s, sys: 11 s, total: 19min 8s
Wall time: 19min 12s


In [8]:

compute_metrics(
    results_file="results/GSAI-ML_LLaDA-8B-Base_256_256_4_1_0.0_generations_testing_fast_dllm_threshold_0.5.json",
    samples_per_problem=10,
    k_values=[1, 5, 10]
)

Loaded 256 generations

--- Answer Format Analysis ---

Format Distribution:
  float     :  242 ( 94.5%)
  int       :    0 (  0.0%)
  list      :    0 (  0.0%)
  string    :   14 (  5.5%)
  none      :    0 (  0.0%)

Example Formats:

  FLOAT examples:
    1. Q: Tom buys a bedroom set for $3000.  He sells his old bedroom for $1000 and uses t...
       A: 200.0
    2. Q: The Smith family is getting ready for summer and needs to have their swimming po...
       A: 826.0

  STRING examples:
    1. Q: Mel uses a 900-watt air conditioner for 8 hours a day. This means that each hour...
       A: 8.1.0
    2. Q: Cayley markets various newspapers to get a commission on each copy sold. He gets...
       A: 15.30.0

--- Computing Pass@k Metrics ---

Problems evaluated: 256
Total samples: 256
Total correct: 151
Correct rate: 0.5898
Accuracy (any correct): 0.5898
Avg correct per problem: 0.59

Pass@k Metrics:
  pass@ 1: 0.5898 (58.98%)
  pass@ 5: 0.0000 (0.00%)
  pass@10: 0.0000 (0.00%)

--- Samp

{'pass_at_k': {1: 0.58984375, 5: 0.0, 10: 0.0},
 'accuracy': 0.58984375,
 'avg_correct_per_problem': 0.58984375,
 'correct_rate': 0.58984375,
 'problems_evaluated': 256,
 'total_correct': 151,
 'total_samples': 256}

In [2]:
%%time

diffusion_model_name = "GSAI-ML/LLaDA-8B-Base"
data = "gsm8k"
num_evals_to_use = 256
few_shot = 4
batch_size = 1
gen_length = 256
diffusion_steps = 256
temperature = 0.0
cfg_scale = 0.0
steps = 256
block_length = 32
remasking = "low_confidence"
alg = "entropy"
alg_temp = 0.0
top_p = 0.95
top_k = None
n_samples = 1
use_cache=True
dual_cache=True
threshold = 0.75
factor = 1.0

metrics = evaluate_fast_dllm(
    diffusion_model_name, data, num_evals_to_use, few_shot, batch_size, gen_length, diffusion_steps, temperature, cfg_scale, steps, block_length, remasking, alg, alg_temp, top_p, top_k, n_samples, use_cache, dual_cache, threshold, factor
    )

Loading checkpoint shards: 100%|██████████| 6/6 [00:00<00:00, 171.11it/s]


Created 4 few-shot examples
evaluating 256 examples
Model type: Base


Evaluating (total samples: 256): 100%|██████████| 256/256 [25:08<00:00,  5.89s/it]


Saved generations to results/GSAI-ML_LLaDA-8B-Base_256_256_4_1_256_0.0_generations_testing_fast_dllm.json
CPU times: user 25min 13s, sys: 12.3 s, total: 25min 25s
Wall time: 25min 30s


In [5]:

compute_metrics(
    results_file="results/GSAI-ML_LLaDA-8B-Base_256_256_4_1_256_0.0_generations_testing_fast_dllm.json",
    samples_per_problem=10,
    k_values=[1, 5, 10]
)

Loaded 256 generations

--- Answer Format Analysis ---

Format Distribution:
  float     :  248 ( 96.9%)
  int       :    0 (  0.0%)
  list      :    0 (  0.0%)
  string    :    8 (  3.1%)
  none      :    0 (  0.0%)

Example Formats:

  FLOAT examples:
    1. Q: Janet had 22 green pens and 10 yellow pens. Then she bought 6 bags of blue pens ...
       A: 98.0
    2. Q: Together, Sofie, Anne, and Fawn have 85 books. If Sofie has 25 more books than A...
       A: 24.0

  STRING examples:
    1. Q: Christina records her mood every day on a calendar. Over the past thirty days of...
       A: <reasoning>
Christina has 12 good days + 8 bad days + 8 neutral days = <<12+8+8=28>>28 days in the first 28 days.
She has 28 days - 8 days = <<28-8=20>>20 days left in the first 28 days.
She has 20 da
    2. Q: Avery needs to buy a 3 piece place setting (dinner & salad plate and a bowl) for...
       A: <reasoning>
She needs 3 place settings per person and is having 12 guests so she needs 3*12 = <<3*1

{'pass_at_k': {1: 0.6796875, 5: 0.0, 10: 0.0},
 'accuracy': 0.6796875,
 'avg_correct_per_problem': 0.6796875,
 'correct_rate': 0.6796875,
 'problems_evaluated': 256,
 'total_correct': 174,
 'total_samples': 256}

In [2]:



compute_metrics(
    results_file="results/Dream-org_Dream-v0-Base-7B_256_256_4_1_256_0.0_generations_testing.json",
    samples_per_problem=10,
    k_values=[1, 5, 10]
)

Loaded 256 generations

--- Answer Format Analysis ---

Format Distribution:
  float     :  248 ( 96.9%)
  int       :    0 (  0.0%)
  list      :    0 (  0.0%)
  string    :    8 (  3.1%)
  none      :    0 (  0.0%)

Example Formats:

  FLOAT examples:
    1. Q: Janet had 22 green pens and 10 yellow pens. Then she bought 6 bags of blue pens ...
       A: 98.0
    2. Q: Together, Sofie, Anne, and Fawn have 85 books. If Sofie has 25 more books than A...
       A: 28.0

  STRING examples:
    1. Q: Frederick is making popsicles to sell and to save money he is making his own pop...
       A: <reasoning>
A 2 x 4 makes 200 sticks and a 2 x 8 makes 400 sticks.
A 2 x 4 costs $4 and a 2 x 8 costs $6.
He has $24 to buy wood for sticks.
He can buy 6 2 x 4s for $24.
He can buy 4 2 x 8s for $24.
H
    2. Q: Two apples fell out of the tree, and one of them landed on Newton's head.  Newto...
       A: <reasoning>
The first two apples that fell out of the tree were picked up by Newton, and he put one

{'pass_at_k': {1: 0.75390625, 5: 0.0, 10: 0.0},
 'accuracy': 0.75390625,
 'avg_correct_per_problem': 0.75390625,
 'correct_rate': 0.75390625,
 'problems_evaluated': 256,
 'total_correct': 193,
 'total_samples': 256}

In [3]:

compute_metrics(
    results_file="results/GSAI-ML_LLaDA-8B-Base_256_256_4_1_256_0.0_generations_testing.json",
    samples_per_problem=10,
    k_values=[1, 5, 10]
)

Loaded 256 generations

--- Answer Format Analysis ---

Format Distribution:
  float     :  244 ( 95.3%)
  int       :    0 (  0.0%)
  list      :    0 (  0.0%)
  string    :   12 (  4.7%)
  none      :    0 (  0.0%)

Example Formats:

  FLOAT examples:
    1. Q: Janet had 22 green pens and 10 yellow pens. Then she bought 6 bags of blue pens ...
       A: 98.0
    2. Q: Together, Sofie, Anne, and Fawn have 85 books. If Sofie has 25 more books than A...
       A: 28.0

  STRING examples:
    1. Q: Christina records her mood every day on a calendar. Over the past thirty days of...
       A: <reasoning>
Christina had 12 good days + 8 bad days + 8 neutral days = <<12+8+8=28>>28 days in the first 24 days.
She had 8 good days + 8 bad days + 8 neutral days = <<8+8+8=24>>24 days in the next 24
    2. Q: Jo has been making face masks. She can make 4 small masks with 2 yards of materi...
       A: 7.67.0

--- Computing Pass@k Metrics ---

Problems evaluated: 256
Total samples: 256
Total correct:

{'pass_at_k': {1: 0.6875, 5: 0.0, 10: 0.0},
 'accuracy': 0.6875,
 'avg_correct_per_problem': 0.6875,
 'correct_rate': 0.6875,
 'problems_evaluated': 256,
 'total_correct': 176,
 'total_samples': 256}

In [2]:
%%time

diffusion_model_name = "GSAI-ML/LLaDA-8B-Base"
data = "gsm8k"
num_evals_to_use = 256
few_shot = 4
batch_size = 1
gen_length = 256
diffusion_steps = 256
temperature = 0.0
cfg_scale = 0.0
steps = 256
block_length = 32
remasking = "low_confidence"
alg = "entropy"
alg_temp = 0.0
top_p = 0.95
top_k = None
n_samples = 1
use_cache=True
dual_cache=True
threshold = 0.9
factor = 1.0

filename = evaluate_fast_dllm(
    diffusion_model_name, data, num_evals_to_use, few_shot, batch_size, gen_length, diffusion_steps, temperature, cfg_scale, steps, block_length, remasking, alg, alg_temp, top_p, top_k, n_samples, use_cache, dual_cache, threshold, factor
    )

compute_metrics(
    results_file=filename,
    samples_per_problem=10,
    k_values=[1, 5, 10]
)

Loading checkpoint shards: 100%|██████████| 6/6 [00:00<00:00, 175.66it/s]


Created 4 few-shot examples
evaluating 256 examples
Model type: Base


Evaluating (total samples: 256): 100%|██████████| 256/256 [28:21<00:00,  6.64s/it]


Saved generations to results/GSAI-ML_LLaDA-8B-Base_256_256_4_1_256_0.0_generations_testing_fast_dllm.json
Loaded 256 generations

--- Answer Format Analysis ---

Format Distribution:
  float     :  248 ( 96.9%)
  int       :    0 (  0.0%)
  list      :    0 (  0.0%)
  string    :    8 (  3.1%)
  none      :    0 (  0.0%)

Example Formats:

  FLOAT examples:
    1. Q: Janet had 22 green pens and 10 yellow pens. Then she bought 6 bags of blue pens ...
       A: 98.0
    2. Q: Together, Sofie, Anne, and Fawn have 85 books. If Sofie has 25 more books than A...
       A: 24.0

  STRING examples:
    1. Q: Christina records her mood every day on a calendar. Over the past thirty days of...
       A: <reasoning>
Christina has 12 good days + 8 bad days + 8 neutral days = <<12+8+8=28>>28 days in the first 28 days.
She has 28 days - 8 days = <<28-8=20>>20 days left in the first 28 days.
She has 20 da
    2. Q: Avery needs to buy a 3 piece place setting (dinner & salad plate and a bowl) for...
   

{'pass_at_k': {1: 0.69140625, 5: 0.0, 10: 0.0},
 'accuracy': 0.69140625,
 'avg_correct_per_problem': 0.69140625,
 'correct_rate': 0.69140625,
 'problems_evaluated': 256,
 'total_correct': 177,
 'total_samples': 256}

In [2]:
%%time

diffusion_model_name = "Dream-org/Dream-v0-Base-7B"
data = "gsm8k"
num_evals_to_use = 256
few_shot = 4
batch_size = 1
gen_length = 256
diffusion_steps = 256
temperature = 0.0
cfg_scale = 0.0
steps = 256
block_length = 32
remasking = "low_confidence"
alg = "entropy"
alg_temp = 0.0
top_p = 0.95
top_k = None
n_samples = 1
use_cache=True
dual_cache=True
threshold = 0.9
factor = 1.0

filename = evaluate_fast_dllm(
    diffusion_model_name, data, num_evals_to_use, few_shot, batch_size, gen_length, diffusion_steps, temperature, cfg_scale, steps, block_length, remasking, alg, alg_temp, top_p, top_k, n_samples, use_cache, dual_cache, threshold, factor
    )

compute_metrics(
    results_file=filename,
    samples_per_problem=10,
    k_values=[1, 5, 10]
)

Loading checkpoint shards: 100%|██████████| 4/4 [00:00<00:00, 103.64it/s]


Created 4 few-shot examples
evaluating 256 examples
Model type: Base


Evaluating (total samples: 256): 100%|██████████| 256/256 [29:29<00:00,  6.91s/it]


Saved generations to results/Dream-org_Dream-v0-Base-7B_256_256_4_1_256_0.0_generations_testing_fast_dllm.json
Loaded 256 generations

--- Answer Format Analysis ---

Format Distribution:
  float     :  249 ( 97.3%)
  int       :    0 (  0.0%)
  list      :    0 (  0.0%)
  string    :    7 (  2.7%)
  none      :    0 (  0.0%)

Example Formats:

  FLOAT examples:
    1. Q: Janet had 22 green pens and 10 yellow pens. Then she bought 6 bags of blue pens ...
       A: 98.0
    2. Q: Together, Sofie, Anne, and Fawn have 85 books. If Sofie has 25 more books than A...
       A: 60.0

  STRING examples:
    1. Q: Christina records her mood every day on a calendar. Over the past thirty days of...
       A: <reasoning>
Christina had 12 + 8 + 8 = <<12+8+8=28>>28 days of good, bad, and neutral moods.
She had 30 - 28 = <<30-28=2>>2 days of good, bad, and neutral moods.
She had 8 + 8 + 8 = <<8+8+8=24>>24 day
    2. Q: As Sally walked to school, she was holding the strings to 25 red balloons, 7 gre..

{'pass_at_k': {1: 0.7578125, 5: 0.0, 10: 0.0},
 'accuracy': 0.7578125,
 'avg_correct_per_problem': 0.7578125,
 'correct_rate': 0.7578125,
 'problems_evaluated': 256,
 'total_correct': 194,
 'total_samples': 256}